File: src/1422.maximum-score-after-splitting-a-string.1537/solution.ts
function maxScore (s: string): number {
    let one = 0;
    for (const c of s) {
        if (c === '1') {
            one++;
        }
    }
    let result = 0;
    let zero = 0;
    for (let i = 0; i < s.length - 1; i++) {
        if (s[i] === '0') {
            zero++;
        } else {
            one--;
        }
        result = Math.max(result, zero + one);
    }
    return result;
}

File: src/1422.maximum-score-after-splitting-a-string.1537/solution.java
class Solution {
    public int maxScore(String s) {
        char[] chars = s.toCharArray();
        int[] count1s = new int[s.length()];
        int result = 0;
        int one = 0;
        for(int i=chars.length-1;i>0;i--){
            if(chars[i] == '1'){
                one++;
            }
            count1s[i] = one;
        }
        int zero = 0;
        for(int i=0;i<chars.length-1;i++){
            if(chars[i] == '0'){
                zero++;
            }
            result = Math.max(result,count1s[i+1]+zero);
        }

        return result;
    }
}